src/plugins/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -329,6 +330,18 @@ export async function saveVisualization(
     delete savedObject.id;
   }
 
+  const {
+    visState: {
+      params: { seriesParams },
+    },
+  } = savedObject;
+  const firstSeriesType = seriesParams && seriesParams[0].type;
+  if (firstSeriesType) {
+    savedObject.visState = produce(savedObject.visState, (draftState) => {
+      draftState.params.type = draftState.type = firstSeriesType;
+    });
```

Comment:
Could do this with an object clone, but using `produce` seemed elegant. Open for opinions.